src/Microsoft.Repl.Tests/ShellTests.cs

```diff
@@ -162,8 +162,8 @@ public async Task RunAsync_WithUnhandledKeyPress_DoesNothing()
 
         private Shell CreateShell(ConsoleKey consoleKey, int caretPosition, string previousCommand, string nextCommand, out CancellationTokenSource cancellationTokenSource)
         {
-            HttpState httpState = new HttpState();
-            var defaultCommandDispatcher = DefaultCommandDispatcher.Create(httpState.GetPrompt, httpState);
+            MockHttpState mockHttpState = new MockHttpState();
```

Comment:
The Http here is a red herring.  In the Microsoft.Repl layer, there's nothing Http specific, and hence no reason to refer to Http.  It's more clear to use something like the [Null Object pattern](https://en.wikipedia.org/wiki/Null_object_pattern) to indicate that this state object has no behavior (which is kinda what you've done but the name is confusing - normally you'd see something like `NullState`).

In this case, we can even go a step further because there is no type required for the state, and also the prompt can be decoupled from the state, so we could just do something like:

```
var defaultCommandDispatcher = DefaultCommandDispatcher.Create(() => string.Empty, new object());
```

Reply:
Updated here:
https://github.com/aspnet/HttpRepl/commit/c4c9fa26b8157d81e7312facf8490e26642a895b